src/bonefish/messages/wamp_message.hpp

```diff
@@ -28,22 +29,47 @@ class wamp_message
 {
 public:
     wamp_message();
-    virtual ~wamp_message();
+    virtual ~wamp_message() = default;
 
-    wamp_message(const wamp_message& other) = delete;
+    wamp_message(const wamp_message&) = delete;
+    wamp_message(wamp_message&&) = delete;
     wamp_message& operator=(wamp_message const&) = delete;
+    wamp_message& operator=(wamp_message&&) = delete;
+
+    std::unique_ptr<msgpack::zone>&& release_zone();
+    const std::unique_ptr<msgpack::zone>& get_zone() const;
```

Comment:
If you're not expecting the caller to modify or move the unique_ptr then you might as well return the actual zone pointer directly. I think C++17 will have a type to use for non-owned pointers, but for now a plain pointer is probably the best we have for this use case.